project-base/storefront/components/Basic/ExtendedNextLink/ExtendedNextLink.tsx

```diff
@@ -67,3 +70,22 @@ export const ExtendedNextLink: FC<ExtendedNextLinkProps> = ({
         </NextLink>
     );
 };
+
+const isHrefExternal = (href: string | UrlObject, baseUrl: string) => {
+    const currentHostname = new URL(baseUrl).hostname;
+
+    if (typeof href === 'object') {
+        const linkHostname = href.hostname;
+        if (!linkHostname) {
+            return false;
+        }
+        return currentHostname !== linkHostname;
```

Comment:
Is the special condition necessary? If `linkHostname` is falsy, it will never match `currentHostname` as that should always be defined.

Reply:
You are absolutely right.